runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonPlugin.java

```diff
@@ -301,6 +302,23 @@ public void onChatMessage(ChatMessage event)
 			cannonPlaced = true;
 			addCounter();
 			cballsLeft = 0;
+
+			final ItemContainer inventory = client.getItemContainer(InventoryID.INVENTORY);
+			if (inventory != null)
+			{
+				int invCballs = inventory.count(ItemID.GRANITE_CANNONBALL) > 0
+						? inventory.count(ItemID.GRANITE_CANNONBALL)
+						: inventory.count(ItemID.CANNONBALL);
```

Comment:
I think this logic causes a bug if the player's cannon is destroyed when containing normal cannonballs, then they bring 30+ granite cannonballs and no normal cannonballs the next time they place it. I can't test it because I don't have any granite cannonballs right now.

Reply:
I dont see how 5bfbff492d4bff1f40aeb3cd411f6766b6a6ba67 doesn't address that.

Reply:
5bfbff4 does address that. Sorry for the mistake.